### PR TITLE
insights: store: add support for recording data points

### PR DESCRIPTION
**Pull request by slimsag**
This PR adds support for the store to record data points.

Stacked on top of #18254

Signed-off-by: Stephen Gutekanst <stephen@sourcegraph.com>

### CODE REVIEW

File: enterprise/internal/insights/store/store.go [hunk 1]
```diff
@@ -140,6 +144,119 @@ func seriesPointsQuery(opts SeriesPointsOpts) *sqlf.Query {
...
+		v.Point.Time,  // time
+		v.Point.Value, // value
+		metadataID,    // metadata_id
+		v.RepoID,      // repo_id
+		repoNameID,    // repo_name_id
+		repoNameID,    // original_repo_name_id
+	))
+}
+
+const upsertRepoNameFmtStr = `
```
Comment (efritz) [inline enterprise/internal/insights/store/store.go:223]: Is there a reason these queries don't have the source/use comment?

File: enterprise/internal/insights/store/store.go [hunk 2]
```diff
@@ -140,6 +144,119 @@ func seriesPointsQuery(opts SeriesPointsOpts) *sqlf.Query {
...
+
+	if (v.RepoName != nil && v.RepoID == nil) || (v.RepoID != nil && v.RepoName == nil) {
+		return errors.New("RepoName and RepoID must be mutually specified")
+	}
+
+	// Upsert the repository name into a separate table, so we get a small ID we can reference
+	// many times from the series_points table without storing the repo name multiple times.
+	var repoNameID *int32
+	if v.RepoName != nil {
+		row := txStore.QueryRow(ctx, sqlf.Sprintf(upsertRepoNameFmtStr, *v.RepoName, *v.RepoName))
```
Comment (efritz) [inline enterprise/internal/insights/store/store.go:188]: You could use `basestore.ScanFirstInt` here

File: enterprise/internal/insights/store/store.go [hunk 3]
```diff
@@ -140,6 +144,119 @@ func seriesPointsQuery(opts SeriesPointsOpts) *sqlf.Query {
...
+		}
+	}
+
+	// Upsert the metadata into a separate table, so we get a small ID we can reference many times
+	// from the series_points table without storing the metadata multiple times.
+	var metadataID *int32
+	if v.Metadata != nil {
+		jsonMetadata, err := json.Marshal(v.Metadata)
+		if err != nil {
+			return errors.Wrap(err, "upserting>encoding metadata")
```
Comment (efritz) [inline enterprise/internal/insights/store/store.go:201]: Is the `>` intentional here?

File: enterprise/internal/insights/store/store.go [hunk 4]
```diff
@@ -140,6 +144,119 @@ func seriesPointsQuery(opts SeriesPointsOpts) *sqlf.Query {
...
+		}
+		row := txStore.QueryRow(ctx, sqlf.Sprintf(upsertMetadataFmtStr, jsonMetadata, jsonMetadata))
```
Comment (efritz) [inline enterprise/internal/insights/store/store.go:203]: Same as above

**Review by efritz — APPROVED** [review]

File: enterprise/internal/insights/store/store.go (hunk 3 above)
Comment (slimsag) [reply enterprise/internal/insights/store/store.go:201]: Yeah I tried to use this to signal it's a sub-operation of inserting.. I"ll use a colon.

File: enterprise/internal/insights/store/store.go (hunk 1 above)
Comment (slimsag) [reply enterprise/internal/insights/store/store.go:223]: Fixed, thanks for catching!

**Pull request merged**